### PR TITLE
Feature: Legal Hold - Legal hold details screen for conversation - Show user's device list

### PR DESCRIPTION
**Pull request by gizemb**
### Issues

Jira: [SQSERVICES-345](https://wearezeta.atlassian.net/browse/SQSERVICES-345)

Given that I am on legal hold details screen for a conversation, 
when I click one of the users on legal hold subjects list,
then I should see user's device list

### Solutions

Open user details screen using existing checks with devices tab on if possible.

#### APK
[Download build #3225](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3225/artifact/build/artifact/wire-dev-PR3214-3225.apk)
[Download build #3228](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3228/artifact/build/artifact/wire-dev-PR3214-3228.apk)
[Download build #3230](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3230/artifact/build/artifact/wire-dev-PR3214-3230.apk)
[Download build #3232](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3232/artifact/build/artifact/wire-dev-PR3214-3232.apk)
[Download build #3234](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3234/artifact/build/artifact/wire-dev-PR3214-3234.apk)
[Download build #3243](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3243/artifact/build/artifact/wire-dev-PR3214-3243.apk)
[Download build #3273](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3273/artifact/build/artifact/wire-dev-PR3214-3273.apk)
[Download build #3275](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3275/artifact/build/artifact/wire-dev-PR3214-3275.apk)
[Download build #3277](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3277/artifact/build/artifact/wire-dev-PR3214-3277.apk)
[Download build #3350](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3350/artifact/build/artifact/wire-dev-PR3214-3350.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/participants/fragments/ParticipantFragment.scala [hunk 1]
```diff
@@ -267,7 +269,9 @@ class ParticipantFragment extends ManagerFragment with ConversationScreenControl
         isTeamMember <- userAccountsController.isTeamMember(userId).head
       } userOpt match {
         case Some(user) if user.connection == ACCEPTED || user.expiresAt.isDefined || isTeamMember =>
-          openUserProfileFragment(SingleParticipantFragment.newInstance(), SingleParticipantFragment.Tag)
+          import SingleParticipantFragment._
```
Comment (mejdoo) [inline app/src/main/scala/com/waz/zclient/participants/fragments/ParticipantFragment.scala:272]: out of curiosity, why you imported this class here, not in top?

Comment (gizemb) [reply app/src/main/scala/com/waz/zclient/participants/fragments/ParticipantFragment.scala:272]: no particular reason. it also works if I move it to the top but that way it is consistent with other `case` blocks. this usage also creates a "scope" I believe (i.e. if you need this import outside this `case` block, you need to import it again), which doesn't matter for us now.

**Review by mejdoo — APPROVED** [review]

File: app/src/main/scala/com/waz/zclient/legalhold/LegalHoldInfoFragment.scala [hunk 2]
```diff
@@ -47,9 +49,11 @@ object LegalHoldInfoFragment {
 
   trait Container {
     val legalHoldUsers: Signal[Seq[UserId]]
+
+    def onLegalHoldSubjectClick(userId: UserId): Unit = {}
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/legalhold/LegalHoldInfoFragment.scala:53]: I know that we use "container" traits in some places in Scala code, so this is not an unknown pattern, but I wonder if it's not simpler here to just access the signal with legal hold users directly, and have an event stream for clicks as well. I wil ltry to write more about it tomorrow.

**Review by makingthematrix — CHANGES_REQUESTED** [review]

File: app/src/main/scala/com/waz/zclient/legalhold/LegalHoldInfoFragment.scala (hunk 2 above)
Comment (gizemb) [reply app/src/main/scala/com/waz/zclient/legalhold/LegalHoldInfoFragment.scala:53]: @mejdoo we couldn't come up with a mutual decision here. can you help us decide? we can align via chat.

Comment (mejdoo) [reply app/src/main/scala/com/waz/zclient/legalhold/LegalHoldInfoFragment.scala:53]: I never used this "container" traits in wire-android but I think it ok to introduce to the Scala codebase new approaches for handling click events. This solution looks to me less decoupled and it's very similar to how click listeners works in Android.

**Review by makingthematrix — APPROVED** [review]

File: app/src/main/scala/com/waz/zclient/legalhold/LegalHoldInfoFragment.scala (hunk 2 above)
Comment (makingthematrix) [reply app/src/main/scala/com/waz/zclient/legalhold/LegalHoldInfoFragment.scala:53]: Okay. Approved.

**Comment by mejdoo** [comment]
@gizemb you can merge this PR :)

**Comment by gizemb** [comment]
> @gizemb you can merge this PR :)

I will make a small change :)

**Pull request merged**